src/screens/MessageScreen.tsx

```diff
@@ -1,37 +1,112 @@
-import React, { useCallback, useLayoutEffect, useMemo } from 'react';
-import { useHasNewMessagesFromUsers, useStyles, useUser } from '../hooks';
+import React, { useCallback, useLayoutEffect, useMemo, useState } from 'react';
+import { useStyles } from '../hooks';
 import { Avatar, Badge, Divider, List } from 'react-native-paper';
-import { TouchableOpacity } from 'react-native';
+import {
+  NativeScrollEvent,
+  NativeSyntheticEvent,
+  TouchableOpacity,
+} from 'react-native';
 import { ScrollView, View } from 'react-native';
 import { createStyles, useIcons } from '../components/BrandConfigProvider';
 import { HomeStackScreenProps } from '../navigators/types';
 import { t } from 'i18next';
-import { compact, orderBy } from 'lodash';
+import { chunk, compact, orderBy } from 'lodash';
 import { useLookupUsers } from '../hooks/Circles/usePrivatePosts';
 import { ActivityIndicatorView } from '../components';
 import { tID } from '../common';
+import { useUnreadMessages } from '../hooks/useUnreadMessages';
+import { useNotificationManager } from '../hooks/useNotificationManager';
+import { useFocusEffect } from '@react-navigation/native';
 
 export function MessageScreen({
   navigation,
   route,
 }: HomeStackScreenProps<'Home/Messages'>) {
   const { recipientsUserIds } = route.params;
-  const { styles } = useStyles(defaultStyles);
-  const { data, isLoading: userLoading } = useUser();
-
   useLayoutEffect(() => {
     navigation.setOptions({
       title: t('messages-title', 'My Messages'),
     });
   }, [navigation]);
 
+  const { styles } = useStyles(defaultStyles);
   const { User } = useIcons();
+  const [scrollIndex, setScrollIndex] = useState<number>(0);
+
+  // Get userIds that have sent new messages
+  const { unreadMessagesUserIds: unreadUserIds } = useUnreadMessages();
```

Comment:
It might be nice to split out some of this logic into it's own hook to separate it from the component/layout logic.

Reply:
👍 Yeah, totally. I'll follow up with a refactor PR in the near future.